clients/bigquery/bigquery.go

```diff
@@ -35,8 +34,33 @@ type Store struct {
 	db.Store
 }
 
-func (s *Store) getTableConfig(tableData *optimization.TableData) (*types.DwhTableConfig, error) {
-	return utils.GetTableConfig(utils.GetTableCfgArgs{
+func (s *Store) PrepareTemporaryTable(tableData *optimization.TableData, tableConfig *types.DwhTableConfig, tempTableName string, additionalSettings types.AdditionalSettings) error {
```

Comment:
Do we need to be doing this here?
```
// Start temporary table creation
	tempAlterTableArgs := ddl.AlterTableArgs{
		Dwh:               s,
		Tc:                tableConfig,
		FqTableName:       fmt.Sprintf("%s_%s", tableData.ToFqName(s.Label(), false, s.config.SharedDestinationConfig.UppercaseEscapedNames, s.config.BigQuery.ProjectID), tableData.TempTableSuffix()),
		CreateTable:       true,
		TemporaryTable:    true,
		ColumnOp:          constants.Add,
		UppercaseEscNames: &s.config.SharedDestinationConfig.UppercaseEscapedNames,
	}

	if err = ddl.AlterTable(tempAlterTableArgs, tableData.ReadOnlyInMemoryCols().GetColumns()...); err != nil {
		return fmt.Errorf("failed to create temp table: %w", err)
	}
```
I can't find out where we're doing that for BigQuery.

Reply:
Here? https://github.com/artie-labs/transfer/blob/8daf5a96c16cb8c3282df9c911861db309fb88ab/clients/bigquery/merge.go#L119-L132